pkg/agent/controller/networkpolicy/fqdn_test.go

```diff
@@ -584,3 +584,85 @@ func TestSyncDirtyRules(t *testing.T) {
 		})
 	}
 }
+
+func TestOnDNSResponse(t *testing.T) {
+	currentTime := time.Now()
+	tests := []struct {
+		name             string
+		existingDNSCache map[string]dnsMeta
+		responseIPs      map[string]ipWithTTL
+		lowestTTL        uint32
+		expectedIPs      map[string]time.Time
+	}{
+		{
+			name: "new IP added and old IP retained",
+			existingDNSCache: map[string]dnsMeta{
+				"fqdn-test-pod.lfx.test": {
+					responseIPs: map[string]ipWithTTL{
+						// sample IP with some TTL
+						"192.0.2.1": {ip: net.ParseIP("192.0.2.1"), expirationTime: currentTime.Add(10 * time.Second)},
+						// sample IP with time simulating expired time, i thought that using negative time will
+						// simulate expired time as it will always equate  to before when compared to any time.
+						"192.0.2.2": {ip: net.ParseIP("192.0.2.2"), expirationTime: currentTime.Add(-1 * time.Second)},
+					},
+				},
+			},
+			responseIPs: map[string]ipWithTTL{
+				// we get new IP
+				"192.0.2.3": {ip: net.ParseIP("192.0.2.3"), expirationTime: currentTime.Add(10 * time.Second)},
+				// and an exisiting IP
+				"192.0.2.1": {ip: net.ParseIP("192.0.2.1"), expirationTime: currentTime.Add(10 * time.Second)},
```

Comment:
isn't the test more useful if the new TTL for this IP is different from the current one? If the new TTL in the response is greater than the existing one, the new one should be sued. If it is less than the existing one, the current one should remain in use.